lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs

```diff
@@ -722,3 +729,61 @@ hex = convertToBase Base16
 -- | Decode a 'ByteString' from base16
 fromHex :: ByteArray bout => ByteString -> Either String bout
 fromHex = convertFromBase Base16
+
+data ErrUnXPrvStripPub
+    = ErrNoRoundtripMismatch
+      -- ^ The resulting bytestring would have been unable to roundtrip using
+      -- @xPrvFromStrippedPubXPrv@. Most likely because the input @XPrv@ was
+      -- encrypted.
+    | ErrUnexpectedRoundtripError
+      -- ^ Something unexpected went wrong.
+    deriving (Eq, Show)
+
+-- | Extracts the raw xprv bytes as @private key <> chain code@, i.e. without
+-- the public key.
+--
+-- Returns @Left@ if the resulting bytestring fails to roundtrip. Most likely
+-- because the provided key was encrypted.
+unXPrvStripPub :: XPrv -> Either ErrUnXPrvStripPub ByteString
+unXPrvStripPub k = do
+    let res = stripPub . unXPrv $ k
+
+    -- Check that it roundtrips.
+    case (fmap unXPrv . xPrvFromStrippedPubXPrv $ res) of
+            Right bytes
+                | bytes == unXPrv k -> Right res
+                | otherwise         -> Left ErrNoRoundtripMismatch
+            Left  _                 -> Left ErrUnexpectedRoundtripError
+  where
+    -- Converts  prv <> pub <> cc
+    -- To        prv <>        cc
+    stripPub :: ByteString -> ByteString
+    stripPub xprv' = prv <> chainCode
+      where
+        (prv, rest) = BS.splitAt 64 xprv'
+        (_pub, chainCode) = BS.splitAt 32 rest
+
+data ErrXPrvFromStrippedPubXPrv
+    = ErrInputLengthMismatch Int Int -- ^ Expected, Actual
+    | ErrInternalError String
+    deriving (Eq, Show)
+
+-- Create a @XPrv@ from a prv <> cc bytestring.
+xPrvFromStrippedPubXPrv :: ByteString -> Either ErrXPrvFromStrippedPubXPrv XPrv
+xPrvFromStrippedPubXPrv x =
+    let
+        res = toXPrv $ CC.encryptedCreateDirectWithTweak x pass
```

Comment:
Related thread https://github.com/input-output-hk/cardano-wallet/pull/1321#discussion_r382525139

Reply:
The failure seems related to child keys using `DerivationScheme1` more than it seems related to `Random` keys
```
cardano-wallet-jormungandr key root --wallet-style icarus express theme celery coral permit dizzy nothing stadium lemon tube olive brass dutch find mask
68f0cb3d83b5278f0b4c9c4a4ab50e49aef13f348ceafaf8257168fd8f3b894fa47646b6e206864404f3208b7dee1e71cd16096ac9205d9dd5250ae0e963dd79411337bda4d5bc0216462480b809824ffb48f17e08d95ab9f1b91d391e48e66b
cardano-wallet-jormungandr key child --path "0" 68f0cb3d83b5278f0b4c9c4a4ab50e49aef13f348ceafaf8257168fd8f3b894fa47646b6e206864404f3208b7dee1e71cd16096ac9205d9dd5250ae0e963dd79411337bda4d5bc0216462480b809824ffb48f17e08d95ab9f1b91d391e48e66b --wallet-style random | jcli key from-bytes --type ed25519bip32
invalid secret key
  |-> Invalid Secret Key structure
```

Reply:
With b02caf4e5c9373d7145684d2784305390bfbf7af we are using `CC.encryptedCreateDirectWithoutTweak`, and there is no problem.